src/board.js

```diff
@@ -9,22 +9,232 @@
+          return neighbor !== undefined && neighbor.color === Board.EMPTY;
+        });
+        var isKillingOwnGroup = ownSurroundingGroups.every(function(group) {
+          return group.liberties.size === 1;
+        }) && emptyNeighbors.length === 0;
+        if (isDirectlySurroundedByEnemy || isKillingOwnGroup) {
+          // Undo the insertion of a stone.
+          self.set(x, y, Board.EMPTY);
+          return false;
+        }
+      }
+
+      // Group merges.
+      var currentGroup = self.mergeGroups(ownSurroundingGroups);
+      currentGroup.addPoint(point);
+      ownSurroundingGroups.forEach(function(group) {
+        self.groups.delete(group);
+      });
+      self.groups.add(currentGroup);
+
+      enemySurroundingGroups.forEach(function(group) {
+        group.liberties.delete(point);
+      });
+
+      capturedEnemyGroups.forEach(function(group) {
+        self.groups.delete(group);
+        group.points.forEach(function(point) {
+          point.group = null;
+          point.color = Board.EMPTY;
+          // This point's stone was captured. Add the liberties to its
+          // neighbors.
+          self.surrounding(point.x, point.y).forEach(function(neighbor) {
+            if (neighbor !== undefined && neighbor.color !== Board.EMPTY) {
+              neighbor.group.liberties.add(point);
+            }
+          });
+        });
+      });
+
+      if (self.nextPlayingColor === Board.BLACK) {
+        self.nextPlayingColor = Board.WHITE;
+      } else {
+        self.nextPlayingColor = Board.BLACK;
+      }
+      return true;
+    },
+
+    toString: function() {
+      var rows = "  ";
+      for (var i = 0; i < this.size; i++) {
+        rows += rowNameFromNum(i);
+      }
+      rows += "\n ┌";
+      for (var i = 0; i < this.size; i++) {
+        rows += "─";
+      }
+      rows += "┐\n";
+      for (var y = 0; y < this.size; y++) {
+        rows += rowNameFromNum(y) + "│";
+        for (var x = 0; x < this.size; x++) {
+          var point = this.get(x, y);
+          if (point.color === Board.EMPTY) {
+            rows += " ";
+          } else if (point.color === Board.BLACK) {
+            rows += "●";
+          } else if (point.color === Board.WHITE) {
+            rows += "○";
+          }
+        }
+        rows += "│\n";
+      }
+      rows += " └";
+      for (var i = 0; i < this.size; i++) {
+        rows += "─";
+      }
+      rows += "┘";
+      return rows;
+    },
   };
 
+  Board.EMPTY = 0;
+  Board.BLACK = 1;
+  Board.WHITE = 2;
+
+  function stringFromColor(color) {
+    switch (color) {
+      case Board.EMPTY: return 'Empty';
+      case Board.BLACK: return 'Black';
+      case Board.WHITE: return 'White';
+    }
+  }
+  Board.stringFromColor = stringFromColor;
+
+  function rowNameFromNum(number) {
+    if (number < 26) { return String.fromCharCode(97 + number); }
+    else { return String.fromCharCode(65 + number - 26); }
```

Comment:
Going from `z` to `A` is defined in SGF. Nothing is defined beyond `Z`. I can add a check to refuse positions beyond 51.